schema/updates/2020_08_03_mail-in_service.sql

```diff
@@ -0,0 +1,171 @@
+  -- New enum options - needs review: 
+  'HPLC', 'Robot', 'Rack', 'Grid', 'Solids', 'Powder', 'Peltier', 'Spectroscopy', 'CD Spectroscopy', 'Microscopy', 'Imaging', 'CD Thermal Melt', 'Fixed Energy At Ambient With Robot') DEFAULT NULL,
+*/
+--
+-- Alternative 2: Create an ExperimentKind lookup table. 
+-- This can be added now, and then we can worry about migrating the
+-- DiffractionPlan.experimentKind enum to the new lookup table later, thereby
+-- avoiding to break software that depends on the experimentKind enum
+
+CREATE TABLE ExperimentType (
+  experimentTypeId int unsigned auto_increment PRIMARY KEY,
+  name varchar(100),
+  proposalType varchar(100),
+  active boolean DEFAULT 1 COMMENT '1=active, 0=inactive' 
+);
+
+INSERT INTO ExperimentType (experimentTypeId, name, proposalType)
+  VALUES
+    (1, 'Default', NULL),
+    (2, 'MXPressE', NULL),
+    (3, 'MXPressO', NULL),
+    (4, 'MXPressE_SAD', NULL),
+    (5, 'MXScore', NULL),
+    (6, 'MXPressM', NULL),
+    (7, 'MAD', 'MX'),
+    (8, 'SAD', 'MX'),
+    (9, 'Fixed', NULL),
+    (10, 'Ligand binding', 'MX'),
+    (11, 'Refinement', NULL),
+    (12, 'OSC', 'MX'),
+    (13, 'MAD - Inverse Beam', 'MX'),
+    (14, 'SAD - Inverse Beam', 'MX'),
+    (15, 'MESH', 'MX'),
+    (16, 'XFE', 'MX'),
+    (17, 'Stepped transmission', 'MX'),
+    (18, 'XChem High Symmetry', NULL),
+    (19, 'XChem Low Symmetry', NULL),
+    (20, 'Commissioning', NULL),
+    (21, 'HPLC', 'SCM'),
+    (22, 'Robot', 'SCM'),
+    (23, 'Rack', 'SCM'),
+    (24, 'Grid', 'SCM'),
+    (25, 'Solids', 'SCM'),
+    (26, 'Powder', 'SCM'),
+    (27, 'Peltier', 'SCM'),
+    (28, 'Spectroscopy', 'SCM'),
+    (29, 'CD Spectroscopy', 'SCM'),
+    (30, 'Microscopy', 'SCM'),
+    (31, 'Imaging', 'SCM'),
+    (32, 'CD Thermal Melt', 'SCM'),
+    (33, 'Fixed Energy At Ambient With Robot', 'SCM');
+--
+
+-- 'Column' is a reserved word, so we should probably try to find a different 
+-- name for this table.
+CREATE TABLE `Column` (
+  columnId int unsigned auto_increment PRIMARY KEY,
+  name varchar(100),
+  active boolean DEFAULT 1 COMMENT '1=active, 0=inactive' 
+) COMMENT 'Experimental conditions lookup table for BioSAXS';
+
+INSERT INTO `Column` (columnId, name) 
+  VALUES 
+    (1, 's75'),
+    (2, 's200'),
+    (3, 'superose6'),
+    (4, 'kw402.5'),
+    (5, 'kw403'),
+    (6, 'kw404'),
+    (7, 'kw405'),
+    (8, 'user supplied');
+
+ALTER TABLE DiffractionPlan 
+  ADD columnId int unsigned,
+  ADD robotPlateTemperature float COMMENT 'units: ',
+  ADD exposureTemperature float COMMENT 'units: ',  
+  ADD qMin float COMMENT 'minimum in qRange', 
+  ADD qMax float COMMENT 'maximum in qRange',
+  ADD reductionParametersAveraging enum('All', 'Fastest Dimension', '1D') COMMENT '',
+  ADD CONSTRAINT `DiffractionPlan_ibfk2` FOREIGN KEY (`columnId`) REFERENCES `Column` (`columnId`);
+
+ALTER TABLE BLSample
+  ADD isotropy enum('isotropic', 'anisotropic') DEFAULT NULL;
+
+CREATE TABLE ContainerType (
```

Comment:
Yes, I'm aware that some of this is plate specific. The ContainerType table has been proposed a couple of times, I think, most recently by Stu in internal ticket SCI-9339 where he specifically proposes to include the geometry types from SynchWeb's platetypes.js. I think it makes sense to have this in the database rather than a file, provided that there is some kind of admin interface for it, of course.

I think the idea with proposalType is that it will used during shipment creation to filter so that users can only select a ContainerType that has a proposalType (sm, mx, xpdf, em ...) that has been defined in config.php $bl_types for the kind of beamlines/instruments that exists in the sessions on the proposal on which the shipment is created.

Reply:
This will be really useful, as we dont want scm folk to select from all the puck and plate types.